src/configure/helper/templateParameterHelper.ts

```diff
@@ -130,10 +130,12 @@ export class TemplateParameterHelper {
                         else {
 
                             //handling case senstivity issue of httpApplicationRouting
-                            let addonProfiles = JSON.parse(JSON.stringify(detailedResource.properties.addonProfiles).toLowerCase());
-                            addonProfiles.httpApplicationRouting = addonProfiles.httpapplicationrouting;
-                            delete addonProfiles['httpapplicationrouting'];
-                            detailedResource.properties.addonProfiles = addonProfiles;
+                            if (detailedResource.properties.addonProfiles) {
```

Comment:
here as well add detailedResource.properties.addonProfiles.httpApplicationRouting, otherwise it will fail below if httpApplicationRouting is not present.

Reply:
updated